src/products/entities/product.entity.ts

```diff
@@ -4,18 +4,33 @@ import { Base } from '@shared/entities/base.entity';
 
 @Entity('products')
 export class ProductEntity extends Base {
-  @Column({ type: 'varchar', unique: true, length: 255, nullable: false })
-  name: string;
+  @Column({ type: 'varchar', unique: true, nullable: false })
+  readonly name: string;
 
-  @Column({ type: 'text', length: 1000, nullable: true })
-  description: string;
+  @Column({ type: 'text', nullable: true })
+  readonly description: string;
 
   @Column({ type: 'int', nullable: false })
-  price: number;
+  readonly price: number;
 
   @Column({ type: 'int', nullable: false, default: 0 })
-  stock: number;
+  readonly stock: number;
 
-  @Column({ type: 'varchar', nullable: true, length: 500 })
-  image: string;
+  @Column({ type: 'varchar', nullable: true })
+  readonly image: string;
+
+  constructor(
+    name: ProductEntity['name'],
```

Comment:
Can you explain to me what's this syntax doing?

Reply:
Is assigning the same type as the `ProductEntity.name` has

Reply:
@Darvand ?

Reply:
But how this will solve the "Long parameter list"? If I have an object as param that contains all the attributes, then I have to initiate this object somewhere with all params also. So, I don't really understand how this will solve that

Reply:
Because we are encapsulating all the parameters in an object which will serve the purpose of an interface for the method. That will give us the advantage to reuse that interface whenever we need to call the function.
You can read more about Long Parameter List code smell [here ](https://refactoring.guru/smells/long-parameter-list)